From efc78486a2667d13ed780d84618ec239e5728591 Mon Sep 17 00:00:00 2001
From: Tendai Makumire <121951760+tmakumire@users.noreply.github.com>
Date: Thu, 25 Jan 2024 11:50:21 +0000
Subject: [PATCH] Added AR0521 to CamX build (#2)

Now adds the AR0521 Sensor C++ exposure driver to the CamX build
---
 camx/build/infrastructure/linuxembedded/CMakeLists.txt | 2 ++
 camx/src/hwl/titan17x/camxtitan17xcontext.cpp          | 1 +
 2 files changed, 3 insertions(+)

diff --git a/camx/build/infrastructure/linuxembedded/CMakeLists.txt b/camx/build/infrastructure/linuxembedded/CMakeLists.txt
index 043a01b..1db4bae 100755
--- a/camx/build/infrastructure/linuxembedded/CMakeLists.txt
+++ b/camx/build/infrastructure/linuxembedded/CMakeLists.txt
@@ -130,6 +130,8 @@ add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/node/staticafalgo/build/linuxembe
 add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/node/staticawbalgo/build/linuxembedded ${CAMX_CHICDK_OUT_PATH}/staticawbalgo)
 add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/node/staticpdlibalgo/build/linuxembedded ${CAMX_CHICDK_OUT_PATH}/staticpdlibalgo)
 add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/node/dummyfs/build/linuxembedded ${CAMX_CHICDK_OUT_PATH}/dummyfs)
+
+add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/sensor/ar0521/build/linuxembedded ${CAMX_CHICDK_OUT_PATH}/sensor/ar0521)
 add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/sensor/imx318/build/linuxembedded ${CAMX_CHICDK_OUT_PATH}/sensor/imx318)
 add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/sensor/imx334/build/linuxembedded ${CAMX_CHICDK_OUT_PATH}/sensor/imx334)
 add_subdirectory (${CAMX_CHICDK_OEM_PATH}/qcom/sensor/imx362/build/linuxembedded ${CAMX_CHICDK_OUT_PATH}/sensor/imx362)
diff --git a/camx/src/hwl/titan17x/camxtitan17xcontext.cpp b/camx/src/hwl/titan17x/camxtitan17xcontext.cpp
index 140df79..ecdbf7c 100755
--- a/camx/src/hwl/titan17x/camxtitan17xcontext.cpp
+++ b/camx/src/hwl/titan17x/camxtitan17xcontext.cpp
@@ -427,6 +427,7 @@ static const DimensionCap SensorImageSizes[] =
     { 5184, 3880 },
     { 4624, 3472 }, // 16MP
     { 4096, 2304 }, // 9.4 MP
+    { 2600, 1952 }, // For IMDT AR0521 
     { 1928, 1208 }, // For GMSL ar0231
     /// @todo (CAMX-961) This list is not final, can contain more entries
 };
-- 
2.34.1

